src/pyfmi/fmi.pyx

```diff
@@ -8170,20 +8170,28 @@ cdef class FMUModelME2(FMUModelBase2):
         self.__get_real(v_ref_pt, len_v, v_pt)
 
         if group is not None:
-            if "nominals" in group:
-                nominals = <float*>PyArray_DATA(group["nominals"])
-            else:
-                group["nominals"] = N.empty(len_v, dtype=float)
-                nominals = <float*>PyArray_DATA(group["nominals"])
-                for i in range(len_v):
-                    nominals[i] = self.get_variable_nominal(valueref = v_ref_pt[i])
+            if "nominals" in group: # Re-use extracted nominals
+                nominals = group["nominals"]
+                nominals_pt = <FMIL.fmi2_real_t*>PyArray_DATA(nominals)
+            else: # First time extraction of nominals
+                #If we are using the states, then the nominals should instead be picked up from the C callback function for nominals
+                if self._states_references and len_v == len(self._states_references) and (self._states_references[i] == var_ref[i] for i in range(len_v)):
+                    group["nominals"] = N.array(self.nominal_continuous_states, dtype=float)
+                    nominals = group["nominals"]
+                    nominals_pt = <FMIL.fmi2_real_t*>PyArray_DATA(nominals)
+                else:
+                    group["nominals"] = N.empty(len_v, dtype=float)
+                    nominals = group["nominals"]
+                    nominals_pt = <FMIL.fmi2_real_t*>PyArray_DATA(nominals)
+                    for i in range(len_v):
+                        nominals_pt[i] = self.get_variable_nominal(valueref = v_ref_pt[i])
 
             for i in range(len_v):
-                eps_pt[i] = RUROUND*(max(abs(v_pt[i]), nominals[i]))
+                eps_pt[i] = RUROUND*(max(abs(v_pt[i]), nominals_pt[i]))
         else:
             for i in range(len_v):
-                nominal = self.get_variable_nominal(valueref = v_ref_pt[i])
-                eps_pt[i] = RUROUND*(max(abs(v_pt[i]), nominal))
+                tmp_nominal = self.get_variable_nominal(valueref = v_ref_pt[i])
+                eps_pt[i] = RUROUND*(max(abs(v_pt[i]), tmp_nominal))
```

Comment:
```suggestion
                eps_pt[i] = RUROUND*(
                    max(
                        abs(v_pt[i]), self.get_variable_nominal(valueref = v_ref_pt[i])
                        )
                    )
```
Is tmp_nominal only added for readability? If yes then I woud propose simply changing line 8194 by introducing some line breaks, not necessarily as above but some kind of line break could help here.

Reply:
Thansk for the feedback! I view this as a mathematical expression (not specifically different function invokations) and thus I believe it makes sense to have it on one line.